cyclonedx/model/composition.py

```diff
@@ -0,0 +1,231 @@
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# SPDX-License-Identifier: Apache-2.0
+# Copyright (c) OWASP Foundation. All Rights Reserved.
+from enum import Enum
+from typing import Any, Iterable, Optional
+
+import serializable
+from sortedcontainers import SortedSet
+
+from .._internal.compare import ComparableTuple as _ComparableTuple
+from ..serialization import BomRefHelper
+from .bom_ref import BomRef
+
+
+@serializable.serializable_enum
+class AggregateType(str, Enum):
```

Comment:
the functional test for the Enum is missing.

analog to `TestEnumVulnerabilitySeverity` -- see https://github.com/CycloneDX/cyclonedx-python-lib/blob/6d1bc5b5e3c6cecc8681b7a197d07632819dc994/tests/test_enums.py#L469-L486